securityscanutils/trivy_scanner_test.go

```diff
@@ -0,0 +1,83 @@
+package securityscanutils_test
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/solo-io/go-utils/osutils/executils"
+	. "github.com/solo-io/go-utils/securityscanutils"
+)
+
+var _ = Describe("Trivy Scanner", func() {
+	var (
+		t                         *TrivyScanner
+		inputImage                string
+		inputMarkdownTemplateFile string
+		outputDir, outputFile     string
+		err                       error
+	)
+
+	JustBeforeEach(func() {
+		t = NewTrivyScanner(executils.CombinedOutputWithStatus)
+		inputMarkdownTemplateFile, err = GetTemplateFile(MarkdownTrivyTemplate)
+		Expect(err).NotTo(HaveOccurred())
+		outputDir, err := ioutil.TempDir("", "")
+		Expect(err).NotTo(HaveOccurred())
+		outputFile = filepath.Join(outputDir, "test_report.docgen")
+	})
+
+	JustAfterEach(func() {
+		err := os.RemoveAll(outputDir)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("Finds vulnerabilities", func() {
+		inputImage = "quay.io/solo-io/gloo:1.11.1"
+		completed, vulnFound, err := t.ScanImage(context.TODO(), inputImage, inputMarkdownTemplateFile, outputFile)
+
+		Expect(err).NotTo(HaveOccurred())
+		Expect(completed).To(Equal(true))
+		Expect(vulnFound).To(Equal(true))
+	})
+
+	It("Cannot find Image", func() {
+		inputImage = "quay.io/solo-io/gloo:1.11.13245"
+		completed, vulnFound, err := t.ScanImage(context.TODO(), inputImage, inputMarkdownTemplateFile, outputFile)
+
+		Expect(err).NotTo(HaveOccurred())
+		Expect(completed).To(Equal(false))
+		Expect(vulnFound).To(Equal(false))
+
+	})
+
+	It("Returns error from Exec via Timeout", func() {
+		inputImage = "quay.io/solo-io/gloo:1.11.1"
+		completed, vulnFound, err := t.ScanImage(context.TODO(), "", "", "")
+
+		//Error occurs when all trivy scan arguments are empty
+		Expect(err).To(HaveOccurred())
+		Expect(completed).To(Equal(false))
+		Expect(vulnFound).To(Equal(false))
+
+	})
+
+	FContext("Benchmark", func() {
+		It("Should do repeated scans efficiently", func() {
```

Comment:
It looks like we're scanning three different images
Why do we consider the scans to be repeated and why do we expect them to take <4s?

Reply:
Dont we want to scan multiple images or the same image multiple times? Also do we have a time in mind for how quickly we want these scans to finish? I made a guess based off of (local testing time) * 2.

Reply:
> Dont we want to scan multiple images or the same image multiple times?

I think multiple images is what we're doing both in this test and in practice, but the test name is misleading, implying the same image is scanned multiple times

> Also do we have a time in mind for how quickly we want these scans to finish?

This is sort of what I was asking myself. I'm not sure how much time it should take or what our threshold should be specifically. 
If the previous implementation was an order of magnitude more and we intend to verify that we have reduced time by that order of magnitude we should be able to set a threshold at the lower order of magnitude and conservative enough to be hit consistently. 
A comment indicating what it would have taken previously and what it should take now that we avoid retries might be helpful context

Reply:
Let''s use https://github.com/solo-io/ext-auth-service/blob/master/pkg/config/opa/config_test.go#L232 to introduce some more targeted benchmark testing

Reply:
Added! I included the ranges of response times from my local testing environment rather than each test individually, I can add back the multiple images test no problem as well if we need that, but as ben mentioned do we want to keep this test as trivy isn't something we have control over?

Reply:
Another way to do this that's a little cleaner/idiomatic might be to use a [`Consistently()`](https://onsi.github.io/gomega/#consistently) block, ensuring that we don't flake over a certain period of time rather than a set number of retries

Reply:
But if you look into that and don't think it's preferable I'll defer